pandas/tests/reshape/test_merge.py

```diff
@@ -1546,6 +1546,49 @@ def test_dtype_on_categorical_dates(self):
         result_inner = pd.merge(df, df2, how='inner', on=['date'])
         assert_frame_equal(result_inner, expected_inner)
 
+    def test_merging_with_boolean_cateorical_column(self):
+        df1 = pd.DataFrame({'id': [1, 2, 3, 4],
```

Comment:
Can you add the github issue number here as a comment?  And for the other tests you wrote too?

Reply:
done